pyaedt/modules/Material.py

```diff
@@ -1992,16 +1995,276 @@ def set_magnetic_coercivity(self, value=0, x=1, y=0, z=0):
+            dictionary with one key (single frequency in Hz) and values are lists of points at
+            that specific frequency (BP curve).
+        core_loss_model_type : str, optional
+            Core loss model type. The default value is ``"Electrical Steel"``.
+            Options are ``"Electrical Steel"`` and ``"Power Ferrite"``.
+        thickness : str, optional
+            Thickness provided as the value plus the unit.
+            The default is ``0.5mm``.
+        conductivity : float, optional
+            Material conductivity.
+            The default is ``0``.
+        coefficient_setup : str, optional
+            Core loss unit. The default is ``"w_per_cubic_meter"``.
+            Options are ``"kw_per_cubic_meter"``, ``"w_per_cubic_meter"``, ``"w_per_kg"``,
+            and ``"w_per_lb"``.
+
+
+        Returns
+        -------
+        list
+            List of core loss coefficients.
+            Returns Kh, Kc, and Ke coefficients if the core loss model is ``"Electrical Steel"``.
+            Returns Cm, X, and Y if the core loss model is ``"Power Ferrite"``.
+
+        Examples
+        --------
+        This example shows how to get core loss coefficients for Electrical Steel core loss model.
+
+        >>> from pyaedt import Maxwell3d
+        >>> m3d = Maxwell3d()
+        >>> box = m3d.modeler.create_box([-10, -10, 0], [20, 20, 20], "box_to_split")
+        >>> box.material = "magnesium"
+        >>> coefficients = m3d.materials["magnesium"].get_core_loss_coefficients(
+        ...                                                        points_list_at_freq={60 : [[0, 0], [1, 3], [2, 7]]},
+        ...                                                        thickness="0.5mm",
+        ...                                                        conductivity=0)
+        >>> print(coefficients)
+        >>> m3d.release_desktop(True, True)
+        """
+        if not isinstance(points_list_at_freq, dict):
+            raise TypeError("Points list at frequency must be provided as a dictionary.")
+        if not isinstance(thickness, str):
+            raise TypeError("Thickness must be provided as a string with value and unit.")
+        else:
+            value, unit = decompose_variable_value(thickness)
+            if not is_number(value) and not unit:
+                raise TypeError("Thickness must be provided as a string with value and unit.")
+        props = OrderedDict({})
+        freq_keys = list(points_list_at_freq.keys())
+        for i in range(0, len(freq_keys)):
+            if isinstance(freq_keys[i], str):
+                value, unit = decompose_variable_value(freq_keys[i])
+                if unit != "Hz":
+                    value = unit_converter(values=value, unit_system="Freq", input_units=unit, output_units="Hz")
+                points_list_at_freq[value] = points_list_at_freq[freq_keys[i]]
+                del points_list_at_freq[freq_keys[i]]
```

Comment:
If you iterate the dictionary, it will iterate over the keys():

```suggestion
        for k in points_list_at_freq:
            if isinstance(k, str):
                value, unit = decompose_variable_value(k)
                if unit != "Hz":
                    value = unit_converter(values=value, unit_system="Freq", input_units=unit, output_units="Hz")
                points_list_at_freq[value] = points_list_at_freq[k]
                del points_list_at_freq[k]
```

Reply:
Check also other parts of the code for the same changes.